examples/webrtc/main.go

```diff
@@ -1,123 +1,133 @@
+        // _ "github.com/pion/mediadevices/pkg/driver/audiotest"
+        _ "github.com/pion/mediadevices/pkg/driver/camera"     // This is required to register camera adapter
+        _ "github.com/pion/mediadevices/pkg/driver/microphone" // This is required to register microphone adapter
+)
+
+func main() {
+
+        config := webrtc.Configuration{
+                ICEServers: []webrtc.ICEServer{
+                        {
+                                URLs: []string{"stun:stun.l.google.com:19302"},
+                        },
+                },
+        }
+
+        // Create a new RTCPeerConnection
+        x264Params, err := x264.NewParams()
+        if err != nil {
+                panic(err)
+        }
+        x264Params.BitRate = 500_000 // 500kbps
+
+        opusParams, err := opus.NewParams()
+        if err != nil {
+                panic(err)
+        }
+
+        codecSelector := mediadevices.NewCodecSelector(
+                mediadevices.WithVideoEncoders(&x264Params),
+                mediadevices.WithAudioEncoders(&opusParams),
+        )
+
+        mediaEngine := webrtc.MediaEngine{}
+        codecSelector.Populate(&mediaEngine)
+
+        api := webrtc.NewAPI(webrtc.WithMediaEngine(&mediaEngine))
+        peerConnection, err := api.NewPeerConnection(config)
+        if err != nil {
+                panic(err)
+        }
+
+        // Set the handler for ICE connection state
+        // This will notify you when the peer has connected/disconnected
+        peerConnection.OnICEConnectionStateChange(func(connectionState webrtc.ICEConnectionState) {
+                fmt.Printf("Connection State has changed %s \n", connectionState.String())
+        })
+
+        s, err := mediadevices.GetUserMedia(mediadevices.MediaStreamConstraints{
+                Video: func(c *mediadevices.MediaTrackConstraints) {
+                        c.FrameFormat = prop.FrameFormat(frame.FormatI420)
+                        c.Width = prop.Int(640)
+                        c.Height = prop.Int(480)
+                },
+                Audio: func(c *mediadevices.MediaTrackConstraints) {
+                },
+                Codec: codecSelector,
+        })
+        if err != nil {
+                panic(err)
+        }
+
+        for _, track := range s.GetTracks() {
+                track.OnEnded(func(err error) {
+                        fmt.Printf("Track (ID: %s) ended with error: %v\n",
+                                track.ID(), err)
+                })
+
+                _, err = peerConnection.AddTransceiverFromTrack(track,
+                        webrtc.RtpTransceiverInit{
+                                Direction: webrtc.RTPTransceiverDirectionSendonly,
+                        },
+                )
+                if err != nil {
+                        panic(err)
+                }
+        }
+
+        // Create an offer
+        offer, err := peerConnection.CreateOffer(nil)

```

Comment:
Here I have the Golang program create an offer that the user pastes into the Golang base64 Session Description box in the html file. Then the user pastes the browser's answer into the golang console. At first I thought this was the only way to make the example work, because I only could get the example working by having the golang program offer and browser answer, but now I think I might've just made an error in my previous Html code. I'll investigate real quick.